third_party/gflags/CMakeLists.txt

```diff
@@ -1,36 +1,18 @@
-message(STATUS "Configuring and building gflags")
-if(MSVC)
-    # cmake links dynamic runtime libraries by default in Visual Studio
-    # use the ginkgo's flags to use the same runtime libraries as ginkgo
-    ginkgo_load_git_package(gflags_external
-        "https://github.com/gflags/gflags.git"
-        "f7388c6655e699f777a5a74a3c9880b9cfaabe59"
-        "-DGFLAGS_BUILD_TESTING=OFF" "-DGFLAGS_BUILD_gflags_LIB=OFF"
-        "-DGFLAGS_BUILD_gflags_nothreads_LIB=ON" "-DGFLAGS_BUILD_STATIC_LIBS=ON"
-        "-DGFLAGS_BUILD_PACKAGING=OFF" "-DCMAKE_CXX_FLAGS_DEBUG=${CMAKE_CXX_FLAGS_DEBUG}"
-        "-DCMAKE_CXX_FLAGS_RELEASE=${CMAKE_CXX_FLAGS_RELEASE}" "-DCMAKE_C_FLAGS_DEBUG=${CMAKE_C_FLAGS_DEBUG}"
-        "-DCMAKE_C_FLAGS_RELEASE=${CMAKE_C_FLAGS_RELEASE}")
-else()
-    # There is a weird issue with Intel 19 and c++17 causing a linking error.
-    # Use c++14 instead.
-    set(INTEL19_STD_FIX "")
-    if (CMAKE_CXX_COMPILER_ID STREQUAL "Intel" AND CMAKE_CXX_COMPILER_VERSION MATCHES "19")
-        set(INTEL19_STD_FIX "-DCMAKE_CXX_FLAGS=-std=c++14")
+if(NOT gflags_FOUND)
```

Comment:
So we reverse the current behavior in that case, which we had previously defined to be use ours first except if explicitly requested. I think that eases package management, but might surprise some users. You should document this in the relevant `.md` file(s).

Maybe to make this clearer, we could put all these `if(NOT xxx_FOUND)` in `third_party/CMakeLists.txt` as well?

Reply:
good idea. In which cases did we actually use the existing libraries then?

Reply:
Most of the time that's already in `third_party/CMakeLists.txt`. Adding `AND NOT xxx_FOUND` to the existing `if` conditions should just work.

Reply:
I think `ginkgo_add_subdirectory` did the same thing (only include if `NOT xxx_FOUND`), so this should not change the behavior.